micronaut-mcp-http-server/src/main/java/io/micronaut/mcp/http/server/McpHttpRequest.java

```diff
@@ -15,18 +15,22 @@
  */
 package io.micronaut.mcp.http.server;
 
+import io.micronaut.core.annotation.Introspected;
 import io.micronaut.core.annotation.NonNull;
 import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.annotation.Header;
+import io.micronaut.http.annotation.RequestBean;
 
 /**
  * @see <a href="https://modelcontextprotocol.io/specification/2025-06-18/basic/transports#protocol-version-header">Protocol Version Header</a>
  * @param protocolVersion MCP Version
  * @param sessionId Session ID
  * @param lastEventId Last Event ID
  */
-public record McpHttpRequest(@NonNull String protocolVersion,
-                             @Nullable String sessionId,
-                             @Nullable String lastEventId) {
+@Introspected
```

Comment:
you can declare `@RequestBean` here instead that way you don't have to repeat it everywhere